### PR TITLE
[FIX] l10n_in: Place of supply provision for Indian companies

### PR DESCRIPTION
**Pull request by joanna350**
1. Install [Expenses], [Indian - Accounting], [Accounting] on Apps
2. [Settings] [Users & Companies] [Companies]
- create a company based in India with address there
3. Enter [Expenses], [CREATE]
- create and add an employee w contact info (private address) in India
- set paid by: Company, write an amount, category > [CREATE REPORT]
- [SUBMIT TO MANAGER] [APPROVE] [POST TO JOURNAL ENTRIES]
4. Click on the [Journal Entry] created at the top right corner

Issue: No Place of Supply which hinders the client

Resolve by: providing one by default and enable their further revision

Impacted versions: 16.0 up to master

opw-3168426

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
looks good to me

first version seems also a good solution if we want to keep the required attribute like currently: https://github.com/odoo/odoo/commit/bc36219bcdbf4625ddf4f7e560f2c712496b9629

and maybe we could use move_type instead of adding a new related field journal_type.

note: I think in general it's better to prefix a field with l10n_in_ if it's available only in l10n_in_ module (the alternative would be to add the field in accounting module).

File: addons/l10n_in/views/account_invoice_views.xml [hunk 1]
```diff
@@ -7,8 +7,9 @@
         <field name="arch" type="xml">
             <xpath expr="//field[@name='ref']" position="after">
                 <field name="country_code" invisible="1"/>
+                <field name="l10n_in_journal_type" invisible="1"/>
                 <field name="l10n_in_state_id" domain="[('country_id.code', '=', 'IN')]"
-                    attrs="{'invisible': ['|', ('country_code', '!=', 'IN'), ('move_type', '=', 'entry')], 'required': [('country_code', '=', 'IN'), ('move_type', '!=', 'entry')], 'readonly': [('state', '!=', 'draft')]}"/>
+                    attrs="{'invisible': ['|', ('country_code', '!=', 'IN'), ('move_type', '=', 'entry')], 'required': [('country_code', '=', 'IN'), ('move_type', '!=', 'entry'), ('l10n_in_journal_type', 'in', ('sale', 'purchase'))], 'readonly': [('state', '!=', 'draft')]}"/>
```
Comment (jva-odoo) [inline addons/l10n_in/views/account_invoice_views.xml:12]:
```suggestion
                    attrs="{'invisible': ['|', ('country_code', '!=', 'IN'), ('move_type', '=', 'entry')], 'required': [('country_code', '=', 'IN'), ('move_type', '!=', 'entry'), ('journal_id.type', 'in', ('sale', 'purchase'))], 'readonly': [('state', '!=', 'draft')]}"/>
```

File: addons/l10n_in/models/account_invoice.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ class AccountMove(models.Model):
     l10n_in_shipping_bill_date = fields.Date('Shipping bill date', readonly=True, states={'draft': [('readonly', False)]})
     l10n_in_shipping_port_code_id = fields.Many2one('l10n_in.port.code', 'Port code', readonly=True, states={'draft': [('readonly', False)]})
     l10n_in_reseller_partner_id = fields.Many2one('res.partner', 'Reseller', domain=[('vat', '!=', False)], help="Only Registered Reseller", readonly=True, states={'draft': [('readonly', False)]})
+    l10n_in_journal_type = fields.Selection(related='journal_id.type')
```
Comment (jva-odoo) [inline addons/l10n_in/models/account_invoice.py:29]:
```suggestion
```

**Review by jva-odoo — CHANGES_REQUESTED** [review]
No need to create field for this

File: addons/l10n_in/views/account_invoice_views.xml (hunk 1 above)
Comment (jva-odoo) [reply addons/l10n_in/views/account_invoice_views.xml:12]: @yes, my bad, it only works in the `domain`, not in `attrs`.

**Review by jva-odoo — APPROVED** [review]

**Review by rigr-odoo — APPROVED** [review]

**Comment by jco-odoo** [comment]
@tsb-odoo @qdp-odoo @jva-odoo I don't think it makes sense to have a place of supply on other account moves than sale or purchase.  We do need it there however and with this fix, the logic of the compute and the view are the same.  (and we do not set a field on moves where it is not needed, taking unnecessary space in the database)

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by pimodoo** [comment]
@robodoo r-

Your PR is breaking the stagings: https://runbot.odoo.com/runbot/build/24166456

**Comment by jva-odoo** [comment]
@pimodoo @jco-odoo we need to do something about this we create PR for localization then runbot auto active a test for that localization.
what did you say?

CC: @odoo/rd-runbot

**Comment by jva-odoo** [comment]
@hale-odoo you do one thing that you can remove required and raise error when sending E-invoice, E-waybill, and GST return

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**